api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -210,7 +210,7 @@ func (m *AzureManagedControlPlane) setDefaultDNSPrefix() {
 
 func (m *AzureManagedControlPlane) setDefaultAKSExtensions() {
 	for _, extension := range m.Spec.Extensions {
-		if extension.Plan.Name == "" {
+		if extension.Plan != nil && extension.Plan.Name == "" {
 			extension.Plan.Name = fmt.Sprintf("%s-%s", m.Name, extension.Plan.Product)
```

Comment:
Could it be possible that `extension.Plan.Product` could be empty since `Plan.Product` is now been marked `//+optional` ?
If yes, is it worth to have another default value set ?

Reply:
Yes it can be empty. Some extensions don't have a product field, such as `microsoft.Flux`. Those extensions will need this field to be null.